src/ru/animelib/src/eu/kanade/tachiyomi/animeextension/ru/animelib/Animelib.kt

```diff
@@ -0,0 +1,408 @@
+
+        val animes = animeList.data.map { it.toSAnime() }
+        return AnimesPage(animes, hasNext)
+    }
+
+    override fun popularAnimeRequest(page: Int): Request {
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("anime")
+        url.addQueryParameter("page", page.toString())
+        url.addQueryParameter("site_id[]", "5")
+        url.addQueryParameter("links[]", "")
+
+        return GET(url.build())
+    }
+
+    // =============================== Search ===============================
+    override fun searchAnimeParse(response: Response) = popularAnimeParse(response)
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val searchParams = AnimelibFilters.getSearchParameters(filters)
+
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("anime")
+        url.addQueryParameter("page", page.toString())
+        url.addQueryParameter("site_id[]", "5")
+        url.addQueryParameter("links[]", "")
+
+        searchParams.genres.include.forEach { url.addQueryParameter("genres[]", it) }
+        searchParams.genres.exclude.forEach { url.addQueryParameter("genres_exclude[]", it) }
+        searchParams.format.forEach { url.addQueryParameter("types[]", it) }
+        searchParams.pegi.forEach { url.addQueryParameter("caution[]", it) }
+        searchParams.ongoingStatus.forEach { url.addQueryParameter("status[]", it) }
+
+        url.addQueryParameter("sort_by", searchParams.sortOrder)
+        if (searchParams.sortDirection.isNotEmpty()) {
+            url.addQueryParameter("sort_type", searchParams.sortDirection)
+        }
+
+        url.addQueryParameter("q", query)
+
+        return GET(url.build())
+    }
+
+    override fun getFilterList() = AnimelibFilters.FILTER_LIST
+
+    // =============================== Utils ===============================
+    private fun fetchPreferredVideoServer(): String {
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("constants")
+        url.addQueryParameter("fields[]", "videoServers")
+
+        val videoServerResponse = client.newCall(GET(url.build())).execute()
+        val videoServers = json.decodeFromString<VideoServerData>(videoServerResponse.body.string())
+
+        val serverPreference = preferences.getString(
+            PrefConstants.PREF_SERVER_KEY,
+            PrefConstants.PREF_SERVER_ENTRIES[0],
+        )
+        if (serverPreference.isNullOrEmpty()) {
+            return videoServers.data.videoServers[0].url
+        }
+
+        for (videoServer in videoServers.data.videoServers) {
+            if (videoServer.label == serverPreference) {
+                return videoServer.url
+            }
+        }
+
+        return videoServers.data.videoServers[0].url
+    }
+
+    private fun kodikVideoLink(playerUrl: String?): List<Video> {
+        val videos = mutableListOf<Video>()
+        // TODO extract video from kodik player
```

Comment:
It's technically possible to run the js script in Aniyomi's js engine. (QuickJS)
However, I don't know how the unscramble algorithm works. Maybe it's just an AES encryption and the key is hidden in the js file somehow. That's the most common thing, I think. Then it might be possible to get the key with a js deobfuscator (we also have a built-in library for that (synchrony))

This might be a lot of work tho, so if you don't get anywhere just leave out the kodik videos.

Reply:
Oh. Somehow I missed that Aniyomi can run js script.
There is nothing special about this unscramble algorithm. It just replace every character with another. Here it is
```js
t.replace(/[a-zA-Z]/g, function(e) {
    return String.fromCharCode((e <= "Z" ? 90 : 122) >= (e = e.charCodeAt(0) + 13) ? e : e - 26)
})
```
I was concerned about implementing this algorithm in code, because simply change one number and parsing function become useless. But now with js engine it's no longer a problem.
I will try to play around with it

Reply:
Never seen numbers change myself, but I never tried to parse it before this week. But according to some forum where people discussing this player there was 3 stages. In 2021 script just reversed base64 string. Later in 2023 script change characters from `ABCDEFGHIJKLMNOPQRSTUVWXYZ` to `NOPQRSTUVWXYZABCDEFGHIJKLM` accordingly. And now (from february) it uses this algorithm

Reply:
A Caesar cipher in the year 2024, that's crazy
Then you don't need to run any js code... Just convert the string to chars and add 13 to each one and convert them back to a string...

Reply:
Its alive, but I'm worried about complexity of this solution